argilla/src/argilla/datasets/_export/_hub.py

```diff
@@ -0,0 +1,182 @@
+
+            HfApi().upload_file(
+                path_or_fileobj=_default_settings_path,
+                path_in_repo=self._default_settings_path,
+                repo_id=repo_id,
+                repo_type="dataset",
+                token=kwargs.get("token"),
+            )
+
+            HfApi().upload_file(
+                path_or_fileobj=_default_dataset_path,
+                path_in_repo=self._default_dataset_path,
+                repo_id=repo_id,
+                repo_type="dataset",
+                token=kwargs.get("token"),
+            )
+
+        if generate_card:
+            sample_argilla_record = next(iter(self.records(with_suggestions=True, with_responses=True)))
+            sample_huggingface_record = hfds[0]
+            card = ArgillaDatasetCard.from_template(
+                card_data=DatasetCardData(
+                    size_categories=size_categories_parser(len(hfds)),
+                    tags=["rlfh", "argilla", "human-feedback"],
+                ),
+                repo_id=repo_id,
+                argilla_fields=self.settings.fields,
+                argilla_questions=self.settings.questions,
+                argilla_guidelines=self.settings.guidelines or None,
+                argilla_vectors_settings=self.settings.vectors or None,
+                argilla_metadata_properties=self.settings.metadata,
+                argilla_record=sample_argilla_record.api_model().model_dump_json(),
+                huggingface_record=sample_huggingface_record,
+            )
+            card.push_to_hub(repo_id, repo_type="dataset", token=kwargs.get("token"))
+
+    @classmethod
+    def from_hub(
+        cls: Type["Dataset"],
+        repo_id: str,
+        show_progress: bool = True,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        """Loads a `FeedbackDataset` from the Hugging Face Hub.
+
+        Args:
+            repo_id: the ID of the Hugging Face Hub repo to load the `FeedbackDataset` from.
+            *args: the args to pass to `datasets.Dataset.load_from_hub`.
+            **kwargs: the kwargs to pass to `datasets.Dataset.load_from_hub`.
+
+        Returns:
+            A `FeedbackDataset` loaded from the Hugging Face Hub.
+        """
+        from datasets import DatasetDict, load_dataset
+        from huggingface_hub import hf_hub_download
+
+        with TemporaryDirectory() as tmpdirname:
+            hf_hub_download(
+                repo_id=repo_id,
+                filename="settings.json",
+                repo_type="dataset",
+                local_dir=tmpdirname,
+            )
+            hf_hub_download(
+                repo_id=repo_id,
+                filename="dataset.json",
+                repo_type="dataset",
+                local_dir=tmpdirname,
+            )
+            dataset = cls.from_disk(tmpdirname)
+
+        hfds = load_dataset(repo_id)
+
+        if isinstance(hfds, DatasetDict) and "split" not in kwargs:
```

Comment:
I have been thinking about this before too. Because splits are quite fundamental in ML and the Hugging Face Hub, perhaps we can consider it as a reserved metadata field in future versions.